### ISSUE TITLE
Aria-live region is not announcing the change in the number of books

### ISSUE THREAD

**Issue by courtneymcgee** [NYPL/staff-picks#101]
Using PC/Firefox/Jaws and PC/Chrome/Jaws

**Comment by EdwinGuzman** [comment]
@wlla what browser are you using and what version of Jaws?

**Comment by wlla** [comment]
Okay, interesting development. This _is_ working for me on mobile. I'm using an Android device with Talkback and Firefox. While the change is being announced (politely) only the new number of books is being announced, rather than the full string which should read "# books found". This only partial announcement may be due to how the "100 books found" text is coded rather than any aria-live region issues. I will document that related issue in the QA doc for Courtney to prioritize, but mention it here for context!

**Comment by wlla** [comment]
IE 11
51.0.1 (32-bit)
Chrome 61 (64-bit)

Jaws 15

**Comment by EdwinGuzman** [comment]
@wlla can you try again? I added `role="alert"` and made the whole string one variable that gets updated every time the app re-renders. So hopefully it doesn't only partially announce the text. On dev: http://staff-picks-dev.us-east-1.elasticbeanstalk.com/books-music-dvds/recommendations/best-books/ya

**Comment by EdwinGuzman** [comment]
I was talking to @ricardoom about this earlier today and I would suggest that we rethink removing un-selectable filters from the sidebar. I understand WHY stakeholders would want this feature and it makes sense, but 1) rearranging the list of filter buttons, 2) trying to focus on the one was the just selected/deselected, and 3) having the aria-live read out the number of books all together do not work well.

VoiceOver seems to read perfectly fine the number of books found when selecting/deselecting the first three filters since they do not move. I sure that keeping the list static and not dynamically removing filters will help move this issue along.

**Comment by wlla** [comment]
The aria-live region is still not announcing (not even partially) on desktop.

**Comment by courtneymcgee** [comment]
Want to signal to you, @EdwinGuzman and @wlla and others that I'm thinking about this and know I need to make a call on it today.

Removing un-selectable filters eases some problems but raises others. How do we present a tag that is un-selectable in an accessible way? Part of the reason for removing un-selectable filters was accessibility / screen reader logic...

It appears I need to decide upon one or the other set of problems.

**Comment by courtneymcgee** [comment]
To summarize for interested team members:
- Removing un-selectable tags presents the UX risk that the results of filtering actions are unclear (the changing # of books on the page isn't announced)
- Displaying un-selectable tags presents the UX risk that the act of filtering itself is unclear (would need to discuss treatment of applied, un-applied, and un-selectable tags both visually and for screen readers)

At this stage, displaying un-selectable tags presents the greater risk UX/product-wise. Working through #105 (and #104 to a lesser extent) should help with announcing the # of books, though! Thx to @wlla and @EdwinGuzman for their help.

**Comment by EdwinGuzman** [comment]
Merge the code. This should be okay to close. Let me know.
